pkg/api/types.go

```diff
@@ -1414,6 +1414,38 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	unversioned.TypeMeta `json:",inline"`
+	ObjectMeta           `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// The minimum number of pods that must be available simultaneously.  Exactly
```

Comment:
It occurs to me that it's going to be tricky to infer the correct number of *expected* pods in order to do the multiplication when this is specified as a percentage.  If the RC (or other pod controller) is lagging, we could be off by one or a lot more than one.  To do a better job than that, we'd need knowledge of the controller's specs and its semantics.

Given the distributed-responsibility design of k8s, I'm not sure of a cleaner path forward than just trusting that the count of all the pods we see as the correct number of "expected" pods.

This approach would not work with DaemonSets, though, since the DS controller doesn't create pods when it doesn't believe they'll fit.